src/stores/queueStore.ts

```diff
@@ -313,6 +313,22 @@ export class TaskItemImpl {
     return this.status?.messages || []
   }
 
+  /**
+   * Server-provided creation time in milliseconds, when available.
+   *
+   * Sources:
+   * - Queue: 5th tuple element may be a metadata object with { create_time }.
+   * - History (Cloud V2): Adapter injects create_time into prompt[3].extra_data.
+   */
+  get createTime(): number | undefined {
+    const extra = (this.extraData as any) || {}
+    const fromExtra =
+      typeof extra.create_time === 'number' ? extra.create_time : undefined
+    if (typeof fromExtra === 'number') return fromExtra
```

Comment:
Optional: I don't know for sure, but this looks like it could just use some safe accesses since it ends up as undefined by default.